### PR TITLE
OLS-540: add resource management for OLS components

### PR DESCRIPTION
**Pull request by vbelouso**
## Description

Added the ability to configure [resources](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-units-in-kubernetes) for OLS components (API server, Console Plugin, data collector containers)

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #
- Closes https://issues.redhat.com/browse/OLS-540

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [x] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: api/v1alpha1/olsconfig_types.go [hunk 1]
```diff
@@ -112,7 +112,20 @@ type DeploymentConfig struct {
...
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Data Collector Container"
+	DataCollectorContainer DataCollectorContainerConfig `json:"dataCollector,omitempty"`
+}
+
+type APIContainerConfig struct {
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Resource Requirements",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:resourceRequirements"}
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+}
+
+type DataCollectorContainerConfig struct {
```
Comment (xrajesh) [inline api/v1alpha1/olsconfig_types.go:128]: @vbelouso We also want to configure this for our lightspeed-console-plugin container too.

File: lightspeed-catalog/index.yaml [hunk 2]
```diff
@@ -57,7 +57,7 @@ properties:
           }
         ]
       capabilities: Basic Install
-      createdAt: "2024-04-28T22:18:32Z"
+      createdAt: "2024-05-03T12:33:20Z"
```
Comment (xrajesh) [inline lightspeed-catalog/index.yaml:60]: @vbelouso I would have expected this file to be updated too. If its not I have to check the update-bundle-catalog make target.

Comment (vbelouso) [reply lightspeed-catalog/index.yaml:60]: @xrajesh, I have tested it, and this file will update only if I update the bundle image. 
For example, from the tag `0.0.1` -> `0.0.2`

**Comment by xrajesh** [comment]
@vbelouso I noticed that the catalog update(opm render)  is  using the bundle image `quay.io/openshift-lightspeed/lightspeed-operator-bundle:v0.0.1` - from quay - which does not have the updated CRD yet. When this PR gets merged, it gets updated. Approving this PR , but may need to run the opm render in PROW in the long run.

**Pull request merged**